x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/action_type_form.tsx

```diff
@@ -346,9 +345,8 @@ function getAvailableActionVariables(
   actionGroup?: ActionGroupWithMessageVariables
 ) {
   const transformedActionVariables: ActionVariable[] = transformActionVariables(
-    actionGroup?.omitOptionalMessageVariables
```

Comment:
Would it be simpler to just add `state` to `REQUIRED_ACTION_VARIABLES`? That way we can take advantage of the existing logic for `omitOptionalMessageVariables` which already looks for recovery action group id.

Reply:
Not sure that i fully understood that. `state` is already in `REQUIRED_ACTION_VARIABLES` did you mean to move it to `OPTIONAL_ACTION_VARIABLES`?

And is `omitOptionalMessageVariables` used only when the actionGroup is `recovered`?

Reply:
Oh yes, sorry. I meant remove `state` from `REQUIRED_ACTION_VARIABLES`.

`omitOptionalMessageVariables` is set to true when `actionGroup.id === selectedAlertType.recoveryActionGroup.id` in `alert_form`:

https://github.com/elastic/kibana/blob/e9350afe0b828fe2ee62f1c5d56fde9d4ad553df/x-pack/plugins/triggers_actions_ui/public/application/sections/alert_form/alert_form.tsx#L555-L564

Reply:
This logic already takes any custom recovery action ids into account.

Reply:
Awesome! Pushed the changes, much more cleaner :) Thank you!